src/sentry_envelope.c

```diff
@@ -477,98 +361,18 @@ sentry__envelope_for_each_request(const sentry_envelope_t *envelope,
                 continue;
             }
         }
-
-        envelope_item_type_t type = envelope_item_get_type(item);
-        switch (type) {
-        case ENVELOPE_ITEM_TYPE_EVENT: {
-            req = prepare_http_request(&event_id, ENDPOINT_TYPE_STORE,
-                "application/json", item->payload, item->payload_len, false);
-            if (!req || !callback(req, envelope, data)) {
-                return;
-            }
-            break;
-        }
-        case ENVELOPE_ITEM_TYPE_ATTACHMENT: {
-            attachments[attachment_count++] = item;
-            break;
-        }
-        case ENVELOPE_ITEM_TYPE_MINIDUMP: {
-            minidump = item;
-            break;
-        }
-        case ENVELOPE_ITEM_TYPE_UNKNOWN: {
-            other[other_count++] = item;
-            break;
-        }
-        default:
-            continue;
-        }
+        sentry__envelope_serialize_item_into_stringbuilder(item, &sb);
```

Comment:
Can we split this into two steps with separate testable functions?
 1. Remove rate limited items from this envelope.
 2. Serialize the envelope in one go

Between these, the transport would simply check if the envelope is empty.

Reply:
Oh nevermind, we don’t need to expose envelope internals if we have rate limiting in the envelope module, and http generation in the transport…